packages/core/src/input/InputManager.ts

```diff
@@ -0,0 +1,346 @@
+          ? _tempHitResult.collider.entity
+          : null;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Update pointers info.
+   * @param type - Pointer change type
+   * @param pointerId - Pointer ID
+   * @param pageX - The pageX of Pointer
+   * @param pageY - The pageY of Pointer
+   */
+  private _changePointer(type: PointerChangeType, pointerId: number, pageX?: number, pageY?: number) {
+    const { _pointerIdToIndex, _pointerList, _actPointerCount: lastCount, _input } = this;
+    let idx = _pointerIdToIndex[pointerId];
+    switch (type) {
+      case PointerChangeType.Remove:
+        if (idx === undefined) {
+          return;
+        } else {
+          const nowCount = (this._actPointerCount = lastCount - 1);
+          if (nowCount != 0) {
+            const removedTouch = _pointerList[idx];
+            if (idx !== lastCount) {
+              _pointerList[idx] = _pointerList[lastCount];
+              _pointerList[lastCount] = removedTouch;
+            }
+            _input.pageX = (_input.pageX * lastCount - removedTouch.pageX) / nowCount;
+            _input.pageY = (_input.pageY * lastCount - removedTouch.pageY) / nowCount;
+          }
+          delete _pointerIdToIndex[pointerId];
+        }
+        break;
+      case PointerChangeType.Update:
+        if (idx === undefined) {
+          if (lastCount > 0 && !this._multiTouchEnabled) {
+            return;
+          }
+          let touch: Pointer;
+          if (!_pointerList[lastCount]) {
+            touch = _pointerList[lastCount] = new Pointer(pointerId, pageX, pageY, lastCount);
+          } else {
+            touch = _pointerList[lastCount];
+            touch.pageX = pageX;
+            touch.pageY = pageY;
+            touch.indexInList = lastCount;
+          }
+          _pointerIdToIndex[pointerId] = lastCount;
+          const nowCount = (this._actPointerCount = lastCount + 1);
+          _input.pageX = (_input.pageX * lastCount + pageX) / nowCount;
+          _input.pageY = (_input.pageY * lastCount + pageY) / nowCount;
+        } else {
+          const touch = _pointerList[idx];
+          _input.pageX += (pageX - touch.pageX) / lastCount;
+          _input.pageY += (pageY - touch.pageY) / lastCount;
+          touch.pageX = pageX;
+          touch.pageY = pageY;
+        }
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * Push event types into an ordered queue.
+   * @param type - Pointer event type
+   */
+  private _pushEventList(type: PointerEventType) {
+    const { _eventList } = this;
+    if (_eventList.length == this._eventLen) {
+      _eventList.push(type);
+    } else {
+      _eventList[this._eventLen] = type;
+    }
+    ++this._eventLen;
+  }
+
+  /**
+   * On pointer down.
+   * @param evt - Pointer Event
+   */
+  private _onPointerDown = (evt: PointerEvent) => {
+    this._changePointer(PointerChangeType.Update, evt.pointerId, evt.pageX, evt.pageY);
```

Comment:
Put this code in `_update()` to ignore perform event processing function fragmentation.

Reply:
Done.